app/server/models/incentive.py

```diff
@@ -0,0 +1,152 @@
+    {
+        'transfer_method': 'NFC',
+        'incentive_type': 'PERCENTAGE',
+        'incentive_amount': 100,
+        'incentive_recipient': 'RECIPIENT',
+        'restrictions': [
+            {
+                'style': 'transaction_count',
+                'days': 7,
+                'transactions': 5
+            },
+            {
+                'style': 'currency',
+                'days': 7,
+                'limit': 5000
+            }
+        ],
+    }
+
+    transfer_method: Is the incentive targeting QR or NFC transactions
+    incentive_types: Possible incentive types are PERCENTAGE or FIXED. 
+        - PERCENTAGE - give incentive_recipient a set percentage of transactions
+        - FIXED - give incentive_recipient a fixed amount of money for each transaction
+    incentive_amount: amount or percentage of transaction to give incentive_recipient
+    incentive_recipient: Possible values are RECIPIENT or SENDER
+    restrictions: 
+        - style: transaction_count
+            - days: Rolling window of how many days this incentive can be triggered
+            - transactions: number of transactions the sender user made in that window of days
+        - style: currency
+            - days: Rolling window of how many days this incentive can be triggered
+            - limit: Total spent by the sender user (BEFORE the currently evaluated incentive is applied)
+    """
+
+    __tablename__       = 'incentive'
+    incentive_rules = db.Column(JSON)
+    organisation_id = db.Column(db.Integer, db.ForeignKey(Organisation.id))
+
+    def handle_incentive(self, transfer, method, do_not_transact = False):
+        """
+        Handles insentives-- checks if they are applicable, then makes the insentive payment
+        """
+        # See if the insentive is relevant
+        if self.incentive_rules['transfer_method'] != method or self.incentive_rules['transfer_method'] != 'ANY':
+            return False
+
+        # Get incentive amounts, calculate percentage if relevant
+        if self.incentive_rules['incentive_type'] == 'FIXED':
+            incentive_amount = self.incentive_rules['incentive_amount']
+        elif self.incentive_rules['incentive_type'] == 'PERCENTAGE':
+            incentive_amount = transfer.transfer_amount * float(self.incentive_rules['incentive_amount'] / 100)
+
+        # Determine who gets the cash!
+        if self.incentive_rules['incentive_recipient'] == 'SENDER':
+            incentive_recipient_transfer_account = transfer.sender_transfer_account
+            incentive_recipient_user = transfer.sender_user
+        elif self.incentive_rules['incentive_recipient'] == 'RECIPIENT':
+            incentive_recipient_transfer_account = transfer.recipient_transfer_account
+            incentive_recipient_user = transfer.recipient_user
+
+        # Process restrictions, apply them to sender_user
+        for r in self.incentive_rules['restrictions']:
```

Comment:
could it make more sense to integrate these into the limit code to bring all the limits into one place?

Reply:
I think it depends on what you think we should do about [this](https://github.com/teamsempo/SempoBlockchain/pull/479#discussion_r497099036). To me, I think it makes sense to allow transactions to continue after the incentives have been exhausted. If we do block them though, I can see integrating them into limits. Does that make sense?